lib/utils.js

```diff
@@ -1,22 +1,55 @@
+/**
+ * Breaks down blobs (strings) in half until they fit in
+ * maximum length.
+ * Pass a maximum string length and an optional encoder (
+ * will ensure that even if the string gets encoded, the
+ * resulting string will be smaller than the maximum length),
+ * and the function will give you back another function, which
+ * you can use to break down a blob (string) into an array
+ * of strings.
+ * @param {number} maxLength 
+ * @param {(string) => string} (encode)
+ * @returns {(string) => Array<string>}
+ */
+const blobSplitter = (maxLength, encode = (e) => e) => (blob) => {
+  if (encode(blob).length <= maxLength || blob.length <= 1) {
+    return [blob];
+  } else {
+    const split = blobSplitter(maxLength, encode);
+    const limit = Math.floor(blob.length / 2);
+    const firstChunk = blob.substr(0, limit);
+    const secondChunk = blob.substr(limit);
+    return [...split(firstChunk), ...split(secondChunk)];
```

Comment:
I don't know how widespread destructuring is supported in browsers

Reply:
see above

Reply:
Babel replaces destructuring with [].concat